module_2/gift_certificates_system/repository/src/main/java/com/epam/esm/dao/GiftCertificateDao.java

```diff
@@ -0,0 +1,69 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.entity.GiftCertificate;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Gift certificate DAO layer class.
+ * Works with database.
+ */
+public abstract class GiftCertificateDao extends EntityDao<GiftCertificate> {
+    /**
+     * Create gift certificate to tag relation in the database.
+     *
+     * @param giftCertificateId - Gift certificate ID.
+     * @param tagId             - Tag ID.
+     */
+    public abstract void createGiftCertificateToTagRelation(long giftCertificateId, long tagId);
+
+    /**
+     * Updating fields of gift certificate.
+     *
+     * @param giftCertificate   - Gift certificate.
+     * @return                  - Operation result (Gift certificate updated or not updated)
+     */
+    public abstract boolean update(GiftCertificate giftCertificate);
+
+    /**
+     * Updating fields of gift certificate by gift certificate ID and not null fields.
+     *
+     * @param id                - Gift certificate ID.
+     * @param namedParameters   - Not null fields.
+     * @return                  - Operation result (Gift certificate updated or not updated)
+     */
+    public abstract boolean update(long id, Map<String, Object> namedParameters);
```

Comment:
Passing map of parameters is not the best solution

Reply:
My suggestion is to leave only one update method, which accepts id and entity with updated fields